qa/workunits/rbd/cli_migration.sh

```diff
@@ -22,6 +22,10 @@ cleanup_tempdir() {
     rm -rf ${TEMPDIR}
 }
 
+expect_false() {
```

Comment:
Ideally this is in `qa/standalone/ceph-helpers.sh`  that's being sourced by this script, `cli_generic.sh` and couple of other workunit RBD scripts?  That way those scripts can use this function and not have this same code (e.g., in `cli_generic.sh`). This can be fixed in a later PR.

`qa/standalone/ceph-helpers.sh` has `expect_failure()` and `test_expect_failure()` that is not used by any component and seems to be overengineered? CephFS in its scripts uses its own `expect_failure()` that uses the exact same code that is here in `expect_false()`.

Reply:
I actually don't see what `qa/standalone/ceph-helpers.sh` is sourced for.  Do you?

> `qa/standalone/ceph-helpers.sh` has [...] that [...] and seems to be overengineered

I know that `qa/standalone` framework is used with great success in the RADOS suite with a few dozen tests based on it, but on a couple of occasions when I wanted to pull something out of `qa/standalone/ceph-helpers.sh` I walked away with the same feeling :laughing:

Reply:
> I actually don't see what `qa/standalone/ceph-helpers.sh` is sourced for. Do you?
> 

Quickly went through the script. I don't see where its functions are used. I think we should merge this PR as is. Remove this unnecessary sourcing in a separate PR.

Reply:
I think I got it.  `qa/standalone/ceph-helpers.sh` has this
```
if type xmlstarlet > /dev/null 2>&1; then
    XMLSTARLET=xmlstarlet
elif type xml > /dev/null 2>&1; then
    XMLSTARLET=xml
else
	echo "Missing xmlstarlet binary!"
	exit 1
fi
```
snippet to accommodate SUSE-based distros and one is supposed to use `$XMLSTARLET` variable instead of encoding the binary name directly in the script.  The existing command for fetching the pool name doesn't follow that though and uses the binary name directly and I copied that.  It should be fine though as this compatibility business seems long outdated:
```
Mon Oct 13 08:52:37 UTC 2014 - toms@opensuse.org

- SPEC file changes
  - Added link from /usr/bin/xml to /usr/bin/xmlstarlet as other
    distributions do the same
  - Did the same for the manpage
```

Reply:
> Remove this unnecessary sourcing in a separate PR.

Opened https://github.com/ceph/ceph/pull/59433